channels/layers.py

```diff
@@ -204,13 +219,14 @@ def __init__(self, *args, **kwargs):
         of the waiting coroutines will get the result.
         """
         assert self.valid_channel_name(channel)
-        while True:
-            try:
-                message = self.channels.get(channel, []).pop(0)
-            except IndexError:
-                asyncio.sleep(0.01)
-            else:
-                return message
+        self._clean_expired()
+
+        queue = self.channels.setdefault(channel, asyncio.Queue())
+
+        # Do a plain direct receive
+        _, message = await queue.get()
```

Comment:
I would add something here to clean up the queue entry in the `self.channels` dict if it's now empty.